### PR TITLE
Issue 250: Fixing the reporting of health to be more accurate for apps with health checks.

### PR DESCRIPTION
**Pull request by TrevinTeacutter**
JIRA: https://jira2.cerner.com/browse/NORBERT-250

Also added fixes for the following JIRAs:
* https://jira2.cerner.com/browse/NORBERT-248
* https://jira2.cerner.com/browse/NORBERT-249

### CODE REVIEW

**Review by willgorman — APPROVED** [review]

File: clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/agent/DcosLoadBalancerCachingAgent.groovy [hunk 1]
```diff
@@ -205,7 +205,7 @@ class DcosLoadBalancerCachingAgent implements CachingAgent, AccountAware, OnDema
 
   private List<App> loadLoadBalancers() {
     // Currently not supporting anything but account global load balancers - no associated region.
-    dcosClient.getApps(accountName).apps.findAll {
+    dcosClient.getAppsForNamespace(accountName).apps.findAll {
```
Comment (willgorman) [inline clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/agent/DcosLoadBalancerCachingAgent.groovy:208]: Is this related to the health check + deployments stuff?

Comment (TrevinTeacutter) [reply clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/agent/DcosLoadBalancerCachingAgent.groovy:208]: Ran into clashing method signatures, so I changed the name of the function.

**Review by mtweten — APPROVED** [review]
